core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -378,10 +378,24 @@ private[spark] class SparkSubmit extends Logging {
     }
 
     // Resolve glob path for different resources.
-    args.jars = Option(args.jars).map(resolveGlobPaths(_, hadoopConf)).orNull
-    args.files = Option(args.files).map(resolveGlobPaths(_, hadoopConf)).orNull
-    args.pyFiles = Option(args.pyFiles).map(resolveGlobPaths(_, hadoopConf)).orNull
-    args.archives = Option(args.archives).map(resolveGlobPaths(_, hadoopConf)).orNull
+    val proxyUser = if (args.proxyUser != null) {
+      UserGroupInformation.createProxyUser(args.proxyUser, UserGroupInformation.getCurrentUser)
```

Comment:
It would be better to follow here (SparkSubmit.scala#146) to specify `hadoopConf`.

```scala
      val hadoopConf = SparkHadoopUtil.get.newConfiguration(sparkConf)
      val ugi = if (args.proxyUser != null) {
        UserGroupInformation.createProxyUser(args.proxyUser,
          UserGroupInformation.getCurrentUser(hadoopConf))
      } else {
        UserGroupInformation.getCurrentUser(hadoopConf)
      }
```

Reply:
I would love to stick to that standard! However I was not able to find an implementation of getCurrentUser that takes parameters and I wasn't able to find `getCurrentUser(hadoopConf)` in the repo - https://github.com/apache/spark/search?q=getCurrentUser&unscoped_q=getCurrentUser. Was I looking in the wrong place 🙂?